plugins/woocommerce/includes/admin/helper/class-wc-helper-updater.php

```diff
@@ -157,10 +157,43 @@ public static function transient_update_themes( $transient ) {
 	public static function setup_update_plugins_messages() {
 		foreach ( WC_Helper::get_local_woo_plugins() as $plugin ) {
 			$filename = $plugin['_filename'];
-			add_action( 'in_plugin_update_message-' . $filename, array( __CLASS__, 'add_install_marketplace_plugin_message' ), 10, 2 );
+			if ( WC_Helper::is_site_connected() ) {
```

Comment:
this calls `get_option()` and it'll be the same value on each iteration, therefore i'd put this outside of foreach and assign it to a var